x-pack/test/api_integration/apis/security/license_downgrade.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect/expect.js';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function ({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+
+  describe('Privileges registration', function () {
+    this.tags(['skipCloud']);
+
+    it('privileges are re-registered on license downgrade', async () => {
+      // Verify currently registered privileges for TRIAL license.
+      // If you're adding a privilege to the following, that's great!
+      // If you're removing a privilege, this breaks backwards compatibility
+      // Roles are associated with these privileges, and we shouldn't be removing them in a minor version.
+      const expectedTrialLicenseDiscoverPrivileges = [
```

Comment:
__note:__ we have proper dedicated tests for the privileges in basic and trial licenses, so I'm just using a small part as an indicator.